### PR TITLE
add test coverage for multiple modules

### PR DESCRIPTION
**Pull request by tvieira**
## What does this implement/fix? Explain your changes

This commit creates a structure to run functional and integration tests. It changes tox configuration so the reports are separated by python version. It also downgrade the docker library requirement in order to be compatible with the supported docker version running on RHEL 7.4. A small change had to be done in the [testinstance](scripts/testinstance) in order to be compatible with the docker 2.1.0. 

In order to run these tests we need to have tox installed in the environment and run the following command:

```shell
# for python 2.7
tox -epy27

# for python 3.6
tox -epy36
```

A coverage report is created for each of python environment and, after running the functional tests, you can check the report by opening the following files:

```shell
# for python 2.7
cover/functional/py27/index.html

# for python 3.6
cover/functional/py36/index.html
```

The `cover/` directory was added to the `.gitignore`

## Does this close any currently open issues?

This PR relates to issue #29 and #40.

## More comments

This commit is not a complete test of all modules but an initial set of tests that put us on coverage for some core modules.

### CODE REVIEW

File: tests/functional/test_api.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        """Test api.ClientAPI.token property without parameters"""
+        assert_is_none(self.cli_emptysettings._client)
+        assert_equal(self.cli_emptysettings._token, None)
+        assert_equal(self.cli_emptysettings.token, None)
+
+    def test_clientapi_token_property(self):
+        """Test api.ClientAPI.token property without parameters"""
+        assert_is_none(self.cli_emptysettings._client)
+        assert_equal(self.cli_emptysettings._token, None)
+        assert_equal(self.cli_emptysettings.token, None)
```
Comment (vi-patel) [inline tests/functional/test_api.py]: These two tests (test_clientapi_empty_settings_token_property and test_clientapi_token_property) are identical.

Comment (tvieira) [reply tests/functional/test_api.py]: hmmm, yeah, strange why I've done that. I think I wanted to do what I achieve with the test below. I'm removing it. Thanks Vimal.

**Review by ryankwilliams — APPROVED** [review]
LGTM

**Review by vi-patel — APPROVED** [review]
looks good.

**Pull request merged**